cloudevents/abstract/event.py

```diff
@@ -0,0 +1,94 @@
+import typing
+from typing import TypeVar
+
+
+class CloudEvent:
+    """
+    The CloudEvent Python wrapper contract exposing generically-available
+    properties and APIs.
+
+    Implementations might handle fields and have other APIs exposed but are
+    obliged to follow this contract.
+    """
+
+    @classmethod
+    def create(
+        cls,
+        attributes: typing.Dict[str, typing.Any],
+        data: typing.Optional[typing.Any],
+    ) -> "AnyCloudEvent":
+        """
+        Factory function.
+        We SHOULD NOT use the __init__ function as the factory because
+        we MUST NOT assume how the __init__ function looks across different python
+        frameworks
+        """
+        raise NotImplementedError()
```

Comment:
because it causes issues in the code coverage tests. 
I cannot trigger the line to be called because an abstract method call fails before the function is called

Reply:
Yeah, sure, but that's not a valid reason to make smth not abstract. You can just create a test no-op or error-throwing implementation directly in the test case.

Reply:
I tried testing the code directly but it did not actually executed the `pass` line of the function

```
@abstractmethod
def dostuff():
  pass
``` 
because the abstract method raised an exception before reaching the pass.

I will do the thing that is mentioned here
https://stackoverflow.com/questions/9202723/excluding-abstractproperties-from-coverage-reports